internal/server/singleprocess/poll_application.go

```diff
@@ -184,8 +152,57 @@ func (a *applicationPoll) buildPollJob(
 
 		// TODO define timeout interval based on projects app poll interval
 	}
+	var jobs []*pb.QueueJobRequest
+
+	log.Trace("Determining which target to generate a status report on")
+
+	// Default to poll on the "latest" lifecycle operation, so if there's a
+	// deploy, queue up a status on deploy. If there is latest is release, then queue that too.
+	if latestDeployment.Deployment != nil {
+		baseJobCopy, err := copystructure.Copy(baseJob)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "failed to generate a job to poll deployment status: %s", err)
+		}
+		deploymentJob := baseJobCopy.(*pb.QueueJobRequest)
+		deploymentJob.Job.Operation = &pb.Job_StatusReport{
+			StatusReport: &pb.Job_StatusReportOp{
+				Target: &pb.Job_StatusReportOp_Deployment{
+					Deployment: latestDeployment,
+				},
+			},
+		}
+		// SingletonId so that we only have one poll operation at
+		// any time queued per app/operation.
+		deploymentJob.Job.SingletonId = fmt.Sprintf("appl-poll/%s/deployment", app.Name)
```

Comment:
This was another one of my things, but we should stay consistent and have this be `app-status-poll`
```suggestion
		deploymentJob.Job.SingletonId = fmt.Sprintf("app-status-poll/%s/deployment", app.Name)
```